src/main/java/seedu/address/model/Model.java

```diff
@@ -64,11 +64,17 @@ public interface Model {
     void deletePerson(Person target);
 
     /**
-     * Adds the given person.
+     * Adds the given person and updates the filteredList to show all.
      * {@code person} must not already exist in the address book.
      */
     void addPerson(Person person);
 
+    /**
+     * Adds the given person without updating the filteredList.
+     * {@code person} must not already exist in the address book.
+     */
+    void addPersonForEdit(Person person);
+
```

Comment:
For the regular `addPerson`, it will reset the filter to display all the contacts.
For a user who has filtered by a certain criteria, then edited a person in the list, it might be confusing if the list is suddenly updated, causing the person they were updating to disappear from view.

Therefore `addPersonForEdit` keeps the current filter. Though if the user has filtered by name, then the displayed list will change.
*Rename `addPersonForEdit` to `AddPersonKeepFilter` for clarity.

Reply:
Gotcha!